overlord/snapstate/autorefresh.go

```diff
@@ -149,6 +150,44 @@ func (m *autoRefresh) AtSeed() error {
 	return nil
 }
 
+func (m *autoRefresh) canRefreshRespectingMetered(now, lastRefresh time.Time) (can bool, err error) {
+	can, err = canRefreshOnMeteredConnection(m.state)
+	if err != nil {
+		return false, err
+	}
+	if can {
+		return true, nil
+	}
+
+	// ignore any errors that occurred while checking if we are on a metered
+	// connection
+	metered, _ := IsOnMeteredConnection()
+	if !metered {
+		return true, nil
+	}
+
+	if now.Sub(lastRefresh) >= maxPostponement {
+		// TODO use warnings when the infra becomes available
+		logger.Noticef("Last refresh more than %d days ago, refreshing anyway", int(maxPostponement.Hours()/24))
+		return true, nil
+	}
```

Comment:
These log messages are somewhat worrying. This is precisely preventing the refresh from taking place, which means every time an auto-refresh is attempted we'll get here again and log again. I suggest turning them into debug messages, and cutting all of the logic below and replacing it by a simpler debug message ("Auto refresh disabled on metered connections.").